pkg/controllers/sync/controller.go

```diff
@@ -110,78 +120,31 @@ type SyncController struct {
 	ensureDeletionRecheckDelay    time.Duration
 	cascadingDeletionRecheckDelay time.Duration
 
-	typeConfig *fedcorev1a1.FederatedTypeConfig
-
-	fedAccessor FederatedResourceAccessor
-
-	hostClusterClient genericclient.Client
-
-	controllerHistory history.Interface
-
-	controllerRevisionStore cache.Store
-
-	controllerRevisionController cache.Controller
-
-	revListerSynced cache.InformerSynced
-
-	limitedScope bool
-
-	cascadingDeleteFinalizer string
-
 	metrics stats.Metrics
 
 	logger klog.Logger
 }
 
-// StartSyncController starts a new sync controller for a type config
-func StartSyncController(
+// NewSyncController returns a new sync controller for the configuration
+func NewSyncController(
+	logger klog.Logger,
 	controllerConfig *util.ControllerConfig,
```

Comment:
Would like to deprecate `ControllerConfig` now that we have `ControllerContext`. Could we extract the specific fields required as separate parameters?

Reply:
done